documentation/src/main/asciidoc/performance_guide/performance_guide.asciidoc

```diff
@@ -0,0 +1,232 @@
+= Infinispan {infinispanversion} Performance Guide
+The Infinispan community
+:toc2:
+:icons: font
+:toclevels: 3
+:numbered:
+:stem:
+
+This guide will give you information and tweaks about tuning Infinispan performance (both server and library mode).
+
+== Capacity planning
+
+Infinispan stores data in memory. Depending on operating mode data is stored in serialized form (server mode) or directly
+directly in memory (library mode). Data size can be estimated using sophisticated tools like  http://openjdk.java.net/projects/code-tools/jol/[Java Object Layout]
+and total amount of required memory can be calculated using the following formulas:
+
+[stem]
+.Total Data Set in library mode
+++++
+Total Data Set = Number Of Entries * (Key Size + Value Size + 200 b (Me tadata))
+++++
+
+[stem]
+.Total Data Set in server mode
+++++
+Total Data Set = Number Of Entries * (Serialized Key Size + Serialized Value Size + 200 b (Me tadata))
+++++
+
+
+In case of Local or Replicated mode, all data needs to fit in memory, so calculating the amount of required memory is trivial.
+
+Calculating memory requirements for Distributed mode is slightly more complicated and requires using the following:
+
+[stem]
+.Required memory in Distributed mode
+++++
+Required Memo ry = (Total Data Set)/(Nodes)*(1+(Node Failures)/(Nodes - Node Failures))
```

Comment:
If I see correctly, the changes you're proposing are:

- Replacing `NodeFailures` with `NumberOfOwners` (as a reminder - `NodeFailures = NumberOfOwners - 1`
- Introducing `MinimumNumberOfNodes`, which is equal to `Nodes - NodeFailures`

I have no strong opinion for the former (I'm OK with either `NumberOfOwners` and `NodeFailures`) but I'm against the latter. Often times when you create a system you're thinking about the possible failures and put VMs (or physical machines) in different racks. So I believe system architects or developers are much more familiar with the number of node failures.

You also changed the equation a little bit, so let me try to do some math on concrete numbers:
```
Total Data Set = 10 GB
Number of Owners = 2
Node Failures = Number of Owners - 1 = 2 - 1 = 1
Nodes = 10
MinumumNumberOfNodes = 10 - 1 = 9

Original formula: (10/10) * (1+(1/(10-1))) = 1 * (1+1/9) = 1.11 GB
Dan's formula: 10 * (2+1) / 9 = 3.33 GB
```

I think the amount of required memory is a bit high and I also don't understand the part with `(number of owners + 1)`.... Could you please elaborate more about this? What is the motivation behind this?

//cc @alanfx @mgencur @rvansa